megazords/ios-rust/MozillaTestServices/MozillaTestServicesTests/PlacesTests.swift

```diff
@@ -434,4 +434,135 @@ class PlacesTests: XCTestCase {
+    // as well as the handling of invalid urls.
+    func testMatchUrl() {
+        let db = api.getWriter()
+
+        let toAdd = [
+            // add twice to ensure its frecency is higher
+            "https://www.example.com/123",
+            "https://www.example.com/123",
+            "https://www.example.com/12345",
+            "https://www.mozilla.com/foo/bar/baz",
+            "https://www.mozilla.com/foo/bar/baz",
+            "https://mozilla.com/a1/b2/c3",
+            "https://news.ycombinator.com/",
+        ]
+
+        for url in toAdd {
+            try! db.applyObservation(visitObservation: VisitObservation(url: url, visitType: .link))
+        }
+        // Should use the origin search
+        XCTAssertEqual("https://www.example.com/", try! db.matchUrl(query: "example.com"))
+        XCTAssertEqual("https://www.example.com/", try! db.matchUrl(query: "www.example.com"))
+        XCTAssertEqual("https://www.example.com/", try! db.matchUrl(query: "https://www.example.com"))
+
+        // Not an origin.
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "example.com/"))
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "www.example.com/"))
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "https://www.example.com/"))
+
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "example.com/1"))
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "www.example.com/1"))
+        XCTAssertEqual("https://www.example.com/123", try! db.matchUrl(query: "https://www.example.com/1"))
+
+        XCTAssertEqual("https://www.example.com/12345", try! db.matchUrl(query: "example.com/1234"))
+        XCTAssertEqual("https://www.example.com/12345", try! db.matchUrl(query: "www.example.com/1234"))
+        XCTAssertEqual("https://www.example.com/12345", try! db.matchUrl(query: "https://www.example.com/1234"))
+
+        XCTAssertEqual("https://www.mozilla.com/foo/", try! db.matchUrl(query: "mozilla.com/"))
+        XCTAssertEqual("https://www.mozilla.com/foo/", try! db.matchUrl(query: "mozilla.com/foo"))
+        XCTAssertEqual("https://www.mozilla.com/foo/bar/", try! db.matchUrl(query: "mozilla.com/foo/"))
+        XCTAssertEqual("https://www.mozilla.com/foo/bar/", try! db.matchUrl(query: "mozilla.com/foo/bar"))
+        XCTAssertEqual("https://www.mozilla.com/foo/bar/baz", try! db.matchUrl(query: "mozilla.com/foo/bar/"))
+        XCTAssertEqual("https://www.mozilla.com/foo/bar/baz", try! db.matchUrl(query: "mozilla.com/foo/bar/baz"))
+        // Make sure the www/non-www doesn't confuse it
+        XCTAssertEqual("https://mozilla.com/a1/b2/", try! db.matchUrl(query: "mozilla.com/a1/"))
+
+        // Actual visit had no www
+        XCTAssertEqual(nil, try! db.matchUrl(query: "www.mozilla.com/a1"))
+        XCTAssertEqual("https://news.ycombinator.com/", try! db.matchUrl(query: "news"))
+    }
```

Comment:
heh not sure we need those tests... they are very redundant and we don't really need to test the FFI layer (since it's all UniFFI'ed now!) I copied those over from the android code, but stopped halfway because I realized we actually can probably just get rid of them

Reply:
Yeah, this is a good question - I'm inclined to agree that these tests aren't valuable, and might even be harmful as we refactor (eg, it's just more redundant work if we intentionally change something.) Some "smoke" tests might make sense (eg, to test that we can load the library and open a connection or similar). I guess I feel the same about the docstrings/comments too - there's a real risk they will get out of date and even inconsistent between the kotlin and swift when the canonical source of truth remains rust. The problem here is that now there's not a good place for these docs, other than the UDL, and I'm not sure it makes sense to ask for the docs to be moved there. tl;dr, I'll be fine with whatever you think is best :)

Reply:
That's a really good point with the docs, I think it's best to get rid of them here and make a point of keeping them up to date in the UDL